geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/FilterHelper.scala

```diff
@@ -205,6 +205,39 @@ object FilterHelper {
       }
     }
 
+  /**
+   * Finds the filter with the lowest known cost and returns the rest in the same order they were in.
```

Comment:
is there a particular reason that *rest* portion is not also sorted best to worse?